### PR TITLE
Mbo/us/aaicddc 98 update versions

### PR DESCRIPTION
**Pull request by ataperekiq**
- merge from isle-dc repo and update `.env` files to use latest starter site package, tag and public files tar dump path (based on `sample.env`)
- update codebase from latest starter site (https://github.com/Islandora-Devops/islandora-starter-site) to upgrade to Drupal 10 and PHP 8.1.22
- update all modules to latest versions, except:
    - remove `citation_select` (not secure and no suitable new version)
    - keep `DANSE` at 2.2.13 as there is a bug in 2.2.18 which prevents the subscribe/unsubscribe button from working properly
- replace the patched OIDC Active Directory module with latest version (2.0.0-beta6) and update dependent OIDC module with 3.0.0-alpha2 to support it. This is required to support Drupal 10
- create OIDC connection in the new version
- remove _modules_ directory from root `.gitignore` and create a `.gitignore` into the _modules_ directory to exclude only _contrib_ (previous `.gitignore` was not correctly adding custom modules)
- move `openid_connect_windows_aad_patched` into the custom modules folder
- update Danse Label Override to make it Drupal 10 compatible, as per https://www.drupal.org/project/danse_label_override/issues/3286891

### CODE REVIEW

File: codebase/composer.json [hunk 1]
```diff
@@ -69,27 +74,27 @@
...
-        "islandora/advanced_search": "^2.0.0",
         "islandora/controlled_access_terms": "^2",
-        "islandora/islandora": "^2",
-        "islandora/openseadragon": "^2",
+        "islandora/islandora": "^2.8.1",
+        "islandora/islandora_mirador": "^2",
         "islandora/views_nested_details": "^1.0",
         "library/pdf.js": "^2.4",
-        "mjordan/islandora_workbench_integration": "^1.0"
+        "mjordan/islandora_workbench_integration": "dev-main"
```
Comment (ajithroy-iqvia) [inline codebase/composer.json:97]: Is this the right version?

File: version [hunk 2]
```diff
@@ -1 +1 @@
-0.0.56
+0.0.57
```
Comment (ajithroy-iqvia) [inline version:1]: This is a major change, should we bump the version segments?

Comment (ataperekiq) [reply version:1]: I don't think we need to do it now as we will upgrade it to 1.0.0 this sprint

File: codebase/composer.json (hunk 1 above)
Comment (ataperekiq) [reply codebase/composer.json:97]: assume so as it's the version in the latest starter template, https://github.com/Islandora-Devops/islandora-starter-site/blob/main/composer.json#L68

**Pull request merged**